src/endpointRoutes.js

```diff
@@ -38,6 +40,55 @@ const createEndpointRoute = router => {
   })
 }
 
+const readEndpointRoute = router => {
+  router.get('/endpoints/:endpointId', async (ctx, next) => {
+    const failureMsg = 'Retrieving of endpoint failed: '
+    const endpointId = ctx.params.endpointId
+
+    await readEndpoint(endpointId)
+      .then(endpoint => {
+        if (endpoint) {
+          ctx.status = 200
+          ctx.body = endpoint
+          logger.info(
+            `Endpoint "${endpoint.endpoint.pattern}" with pattern ${endpoint.endpoint.pattern} has been retrieved`
+          )
+        } else {
+          const error = `Endpoint with id ${endpointId} does not exist`
+          ctx.status = 404
+          ctx.body = {error: error}
+          logger.error(`${failureMsg}${error}`)
+        }
+        next()
+      })
+      .catch(error => {
+        ctx.statusCode = 400
+        handleServerError(ctx, failureMsg, error, logger)
```

Comment:
@bradsawadye / @BMartinos was there no way to get around passing the logger as a param?

Reply:
Also, would this always be a 400? Wouldn't this be a 500 say if the Mongo Instance was unavailable? Or would that error come up somewhere else...

Reply:
@MattyJ007 i havent yet had a chance to look into the logger issue. Im sure there must be a way to do this properly without passing the logger through.

On the status. I think we could possible check the error code from mongoose if they provide one, otherwise default to 500 because we cant assume it failed due to a bad request

Reply:
How about 418 :wink:

Reply:
hahaha that could amaybe work :) . `Mongoose needs caffeine to start up properly: 418!!!`

Reply:
mongoose does not return error codes. To cater for Mongo instance not being available I'll wrap everything in a try and catch block

Reply:
@bradsawadye Don't worry about it - I've added an EventListeners on my branch that will log errors when the Mongo instance is unavailable. It will state that no new Endpoints can be added as it's no longer dynamic.